packages/cli/src/commands/generate-abi.ts

```diff
@@ -0,0 +1,66 @@
+import type { CommandModule, InferredOptionTypes } from "yargs";
+import { Hex, createWalletClient, http } from "viem";
+import { getSystems } from "../deploy/getSystems";
+import { getWorldDeploy } from "../deploy/getWorldDeploy";
+import { getRpcUrl } from "@latticexyz/common/foundry";
+import fs from "node:fs/promises";
+import { resourceToLabel } from "@latticexyz/common";
+import { systemToAbi } from "../utils/systemToAbi";
+
+const generateAbiOptions = {
+  worldAddress: { type: "string", required: true, desc: "Verify an existing World at the given address" },
+  profile: { type: "string", desc: "The foundry profile to use" },
+  rpc: {
+    type: "string",
+    desc: "The RPC URL to use. Defaults to the RPC url from the local foundry.toml",
+  },
+} as const;
+
+type Options = InferredOptionTypes<typeof generateAbiOptions>;
+
+const commandModule: CommandModule<Options, Options> = {
+  command: "generate-abi",
+
+  describe: "Generate ABI's for Systems and World based on World address",
+
+  builder(yargs) {
+    return yargs.options(generateAbiOptions);
+  },
+
+  async handler(args) {
+    await generateAbiHandler(args);
+    process.exit(0);
+  },
+};
+
+export async function generateAbiHandler(opts: Options) {
```

Comment:
thoughts on putting this into a separate dir like `packages/cli/src/init` or something to prepare for more of these types of files?

Reply:
What do you mean by "combining" the CLI commands? Like just putting some of this logic in a helper function? I imagine if we have one command that combines the logic of other commands, they all still live in the `commands` directory but share helpers